core/src/main/java/com/linecorp/armeria/client/endpoint/EndpointGroup.java

```diff
@@ -143,6 +144,13 @@ static EndpointGroup of(EndpointSelectionStrategy selectionStrategy,
     @Override
     Endpoint selectNow(ClientRequestContext ctx);
 
+    /**
+     * Returns the timeout to wait until a successful {@link Endpoint} selection.
+     * {@code 0} means {@link #selectNow(ClientRequestContext)} should always return an {@link Endpoint}.
```

Comment:
Could you also document what happens on the initial selection attempt? Will it wait indefinitely?